ext/rugged/rugged_repo.c

```diff
@@ -1058,59 +1058,64 @@ static VALUE rb_git_repo_expand_oids(int argc, VALUE *argv, VALUE self)
 {
 	VALUE rb_result, rb_oids, rb_expected_type;
 
-	git_otype expected_type = GIT_OBJ_ANY;
-
 	git_repository *repo;
-	git_oid oid;
 	git_odb *odb;
-	int i, error;
+	git_odb_expand_id *expand;
+	long i, expand_count;
+	int error;
 
 	Data_Get_Struct(self, git_repository, repo);
-
 	rb_scan_args(argc, argv, "11", &rb_oids, &rb_expected_type);
 
 	Check_Type(rb_oids, T_ARRAY);
-	expected_type = rugged_otype_get(rb_expected_type);
+	expand_count = RARRAY_LEN(rb_oids);
+	expand = alloca(expand_count * sizeof(git_odb_expand_id));
```

Comment:
If someone passes in a lot of entries, and you later fail the `Check_Type(rb_hex, T_STRING);` on the last entry, this is going to leak a looooot of memory. You'll want to check the type (and do all other things that can cause an exception) before you call `xalloca`/``alloca`.